src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String LINE_SKIP_STRING = "type";
+    private static final String SPLIT_VALUE = ",";
+    private static final int EXPECTED_DATA_LENGTH = 3;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int FRUIT_INDEX = 1;
+    private static final int OPERATION_INDEX = 0;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith(LINE_SKIP_STRING)) {
+                continue;
+            }
+            String[] parts = line.split(SPLIT_VALUE);
+            if (parts.length < EXPECTED_DATA_LENGTH) {
+                throw new RuntimeException(
```

Comment:
When throwing a RuntimeException due to invalid input data format, include the line that caused the exception to provide more context for debugging.